### PR TITLE
fix: hide assigned employees from time-off add list (SDK-894)

### PR DESCRIPTION
**Pull request by aaronlee777**
## Summary

Refactors the time-off **Add Employees** screen to be add-only, fixes a long-standing empty-page bug, and pre-empts the "ineligible employees" 422 from the add endpoint by predicting eligibility client-side. Adds a `hideSelectAll` opt-out to the shared DataView and auto-hides select-all when the table has no rows.

## Changes

- **Add-only UX.** Employees already on the policy are filtered out of the selectable list. Removal lives in the policy detail view's employee table, not here. Drops the remove mutation, the "remove and save" confirm dialog, and the read-only balance branch for existing assignees in both `SelectEmployeesTimeOff` and `SelectEmployeesHoliday`.
- **Client-side filter, search, and pagination.** `useSelectEmployeesData` now fetches the full employees list via `useEmployeesListSuspense` (page 1, `per=100`) plus `useSuspenseQueries` for any remaining pages. Filter, substring search, and pagination all run against the in-memory list. This fixes the empty-page bug where a server-paginated request would return a page that the client filtered down to zero rows after recent additions.
- **Hide-future-hire eligibility filter.** The add-employees endpoint rejects employees with a future `hire_date` as "ineligible" without surfacing the per-uuid reason. The SDK now drops these (and any missing `hireDate`) before they're selectable, so the 422 doesn't fire for that case.
- **Empty-state message.** New i18n key `emptyState: "All eligible employees have already been added to this policy."` rendered when the list is empty and no search is active. Also hides the search input in this state (a search-with-no-results still shows the search bar so the user can clear it).
- **Select-all header behavior.** Added an optional `hideSelectAll` prop plumbed through `useDataView` → `DataView` → `DataTable` / `DataCards` → `EmployeeTable` for explicit opt-out. Separately, `DataTable` now auto-hides the select-all header when `data.length === 0` (parity with `DataCards`). Per-row checkboxes are unaffected. SelectEmployees does **not** set `hideSelectAll` — the header renders when there are rows and is gone when the table is empty.
- **i18n.** Drops unused `removeConfirmDialog.*` and `errors.removeEmployeesFailed` keys.
- **Test coverage.** Direct unit tests for `hideSelectAll` (DataTable + DataCards), `hideSearch` (EmployeeTable), `isStartedByToday` edge cases (undefined, empty, today exact, ±1 day, far future), and a multi-page integration test that exercises the `x-total-pages` header branch and `useSuspenseQueries` merge.

## Related

- Jira ticket: [SDK-894](https://gustohq.atlassian.net/browse/SDK-894)

## Testing

- \`npm run test -- --run src/components/TimeOff/TimeOffManagement/SelectEmployees src/components/TimeOff/shared/EmployeeTable src/components/Common/DataView\`
- Storybook: \`Domain/TimeOff/SelectEmployees\` — verify Default, PartialSelection, AllSelected, WithReassignmentWarning, WithPagination, WithCarryOver, SearchFiltered, EmptySearchResults, UnlimitedPolicy.
- Manual:
  1. Create a fresh time-off policy with no employees on it. Open Add Employees → verify the list shows all eligible employees minus anyone with a future hire date or no hire date.
  2. Select all on page 1, submit. Reopen Add Employees. The remaining unassigned employees should be visible immediately (no empty page).
  3. With every eligible employee already on the policy, open Add Employees. The table should show the empty-state message, no search bar, and no select-all checkbox.
  4. Type a query that matches nothing. The search bar stays, "no search results" message renders.
  5. With rows present, confirm the select-all header checkbox is visible and functional.

## Notes for reviewers

- This is the first table in the SDK with full client-side pagination. The tradeoff: companies with >100 employees will trigger N parallel \`useSuspenseQueries\` requests on mount (gws-flows caps \`per\` at 100). No explicit concurrency cap, but the browser's per-origin connection limit (~6) acts as a natural ceiling. Documented inline with pointers to escape hatches (server-side eligibility filter, fetch-as-you-scroll) if this flow ever needs to support thousands of employees.
- The \`hideSelectAll\` prop is a small, additive expansion to \`DataView\` (not a breaking change). Documented in JSDoc on \`BaseDataViewProps\`.

[SDK-894]: https://gustohq.atlassian.net/browse/SDK-894?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/TimeOff/TimeOffManagement/SelectEmployees/useSelectEmployeesData.ts [hunk 1]
```diff
@@ -1,47 +1,134 @@
...
+    queries: Array.from({ length: Math.max(0, totalServerPages - 1) }, (_, i) =>
+      buildEmployeesListQuery(gustoClient, {
+        companyId,
+        terminated: false,
+        page: i + 2,
+        per: SERVER_MAX_PER_PAGE,
+        include: [Include.AllCompensations],
+      }),
+    ),
+  })
```
[the remark this answers is not in the document]
Comment (aaronlee777) [reply src/components/TimeOff/TimeOffManagement/SelectEmployees/useSelectEmployeesData.ts:61]: Not actionable with client side pagination. Comment left in code

**Comment by jeffredodd** [comment]
@cursoragent Can you address our failed test in CI here?

**Comment by jeffredodd** [comment]
@cursoragent CI / test still is nto passing

**Review by krisxcrash — APPROVED** [review]
